tests/ut/ut_scargo_publish.py

```diff
@@ -0,0 +1,197 @@
+import os
+import subprocess
+from pathlib import Path
+from unittest.mock import MagicMock, patch
+
+import pytest
+from _pytest.logging import LogCaptureFixture
+
+from scargo.commands.publish import (
+    conan_add_conancenter,
+    conan_add_remote,
+    conan_add_user,
+    conan_clean_remote,
+    scargo_publish,
+)
+from scargo.config import Config
+from tests.ut.utils import get_test_project_config
+
+
+@pytest.fixture
+def scargo_publish_test_setup(
+    monkeypatch: pytest.MonkeyPatch, tmp_path: Path
+) -> Config:
+    os.chdir(tmp_path)
+    monkeypatch.setattr("scargo.commands.publish.get_project_root", lambda: tmp_path)
+
+    test_project_config = get_test_project_config()
+    test_project_config.conan.repo["remote_repo_name_1"] = "https://some-url.io"
+    test_project_config.conan.repo["remote_repo_name_2"] = "https://some-url.io"
+
+    monkeypatch.setattr(
+        "scargo.commands.publish.prepare_config",
+        lambda: test_project_config,
+    )
+
+    return test_project_config
+
+
+def test_publish(
+    scargo_publish_test_setup: Config,
+    mock_subprocess_check_call: MagicMock,
+) -> None:
+    # ARRANGE
+    repo_name = "repo_name"
+
+    # ACT
+    scargo_publish(repo_name)
+
+    # ASSERT
+    project_config = scargo_publish_test_setup
+    project_name = project_config.project.name
+    version = project_config.project.version
+
+    all_called_cmds = [cmd.args[0] for cmd in mock_subprocess_check_call.call_args_list]
+    if any(type(cmd) == list for cmd in all_called_cmds):
+        all_called_cmds = [
+            " ".join(cmd) if type(cmd) == list else cmd for cmd in all_called_cmds
+        ]
+
+    conan_clean_cmd = "conan remote clean"
+    conan_add_remote_cmds = [
+        f"conan remote add {repo_name} {repo_url}"
+        for repo_name, repo_url in project_config.conan.repo.items()
+    ]
+    conan_add_conacenter_cmd = "conan remote add conancenter https://center.conan.io"
+    conan_export_cmd = "conan export-pkg . -f"
+    conan_upload_cmd = (
+        f"conan upload {project_name}/{version} -r {repo_name} --all --confirm"
+    )
+
+    assert conan_clean_cmd in all_called_cmds
+    assert all(cmd in all_called_cmds for cmd in conan_add_remote_cmds)
+    assert conan_add_conacenter_cmd in all_called_cmds
+    assert conan_export_cmd in all_called_cmds
+    assert conan_upload_cmd in all_called_cmds
```

Comment:
you're asserting that all commands are present, but you're ignoring their order (which seems to crucial) and any unwanted commands that may be called